client/cmd/dexcctl/main.go

```diff
@@ -26,7 +26,11 @@ const (
 	listCmdMessage  = "Specify -l to list available commands"
 )
 
-var version = semver{major: 0, minor: 3, patch: 0}
+// version is the dexcctl version and should correspond to the rpcclient's
+// version. rpcclients with a higher minor are newer with newer features while
+// a higher major indicates breaking changes and dexcctl should be updated
+// before attempting to communicate.
+var version = semver{major: 0, minor: 4, patch: 0}
```

Comment:
I'm not sure about this.  I was following the is the app product version (dexcctl v0.3 for the upcoming release), while the RPC protocol version is the rpcSemverMinor stuff.

Reply:
That makes sense.

But how will one know this dexcctl will work with the dexc rpcserver? I guess it's reasonable to just require the same release version, although there isn't a way to check from the rpcserver...

Reply:
Maybe I'll try to find a solution in another pr, if it is a problem.

Reply:
It's the same with dcrctl.  It doesn't check protocol version, it just tries cause it's a single shot http post.

> I guess it's reasonable to just require the same release version, although there isn't a way to check from dexcctl...

I don't follow the above.  What can't be checked?

Reply:
The dexc version. dexcctl could be build from a different commit than dexc. So you don't know what version dexc is from rpc commands. There is the rpc version, but there's nothing atm for the user to know from that. So I have version 0.1.0, but I don't know what version dexcctl supports.

Reply:
Gotcha.  That's basically the situation when someone uses say rpcclient to connect to either dcrd or dcrwallet - they have t code what versions are acceptable.

But with dcrctl and dexcctl, being a single shot http post, I'm not sure how to use that info internally (request version prior to every user request?) or if we'd even want to restrict when things might work even out of some predefined range.

Reply:
Since an old dexcctl may work with a new dexc, I would like to put like, a supported dexc rpc version return to dexcctl -V. And this would return 0.1.0 atm because that`s what the rpcserver version is at. And a user would know it supports up to rpc version "0.##.##" Assuming it has users.

Reply:
Makes sense.  That could happen by either introducing supported proto version consts in the dexcttl main package or exporting the ones from client/rpcserver?  And then beyond that it's the user's responsibility to use an rpc server with compatible version, as whatever they deem to be compatible.

Reply:
There have been many times I've used dcrctl and been met with error that left me confused until I realized I needed to update dcrctl.

Reply:
> But with dcrctl and dexcctl, being a single shot http post, I'm not sure how to use that info internally (request version prior to every user request?)

I think I tried to sneak this in when I fist made the thing, but I would be happy with just a way to check manually, like with an extra -V version.

Reply:
> The onus is on the user but they do have a way to check before getting an error.

Agree with this.  I'd rather not cripple dexcctl by being over-restrictive.  A major version bump might actually only affect one RPC, and the tool would otherwise work.

Reply:
#1151